core/src/main/java/org/apache/druid/utils/CollectionUtils.java

```diff
@@ -68,6 +71,17 @@ public int size()
     };
   }
 
+  /**
+   * Returns a transformed map from the given input map where the value is modified based on the given valueMapper
+   * function.
+   */
+  public static <K, V, V2> Map<K, V2> transformValues(Map<K, V> map, Function<V, V2> valueMapper)
```

Comment:
Please use this method elsewhere. There are at least places in `IndexTask`, `TaskQueue` (three places), `KafkaSupervisor`, etc. where it can be used.

It doesn't make sense to add a utility method and not to use it.

To find all such places, use the following [structural search](https://github.com/apache/incubator-druid/blob/master/dev/teamcity.md#creating-a-custom-inspection-from-a-structural-search-pattern):
```
$x$.entrySet().stream().collect($y$)
```

Reply:
Also, don't call the method "transformValues" because it collides with Guava's [`Maps.transformValues()`](https://google.github.io/guava/releases/23.0/api/docs/com/google/common/collect/Maps.html#transformValues-java.util.Map-com.google.common.base.Function-) which returns a lazy map. This would be a source of confusion.

Reply:
You can actually enforce usage of the new method using by adding these two Structural Search patterns:
```
$x$.entrySet().stream().collect(Collectors.toMap(Entry::getKey, $y$))
```
and
```
$x$.entrySet().stream().collect(Collectors.toMap($k$ -> $k$.getKey(), $y$))
```

Reply:
@leventov please note that this is the only remaining release blocker. I don't think any of your comments here should be addressed in this PR. They are out of scope and should be done separately.

Reply:
I'll rename the method, but will address the rest in a follow-up PR after this one is merged ?